support-frontend/assets/pages/[countryGroupId]/components/paymentFields.ts

```diff
@@ -0,0 +1,178 @@
+	}
+}
+
+const getStripePaymentFields = async (
+	stripe: Stripe | null,
+	stripeElements: StripeElements | null,
+	isTestUser: boolean,
+	stripePublicKey: string,
+	recaptchaToken: string | undefined,
+): Promise<RegularPaymentFields | undefined> => {
+	const cardElement = stripeElements?.getElement(CardNumberElement);
+	if (stripe && cardElement && recaptchaToken) {
+		const stripeClientSecret = await stripeCreateSetupIntentRecaptcha(
+			isTestUser,
+			stripePublicKey,
+			recaptchaToken,
+		);
+
+		const stripeIntentResult = await stripe.confirmCardSetup(
+			stripeClientSecret,
+			{
+				payment_method: {
+					card: cardElement,
+				},
+			},
+		);
+
+		if (stripeIntentResult.error) {
+			throw new FormSubmissionError(
+				'There was an issue with your card details.',
+				appropriateErrorMessage(stripeIntentResult.error.decline_code ?? ''),
+			);
+			console.error(stripeIntentResult.error);
+		} else if (stripeIntentResult.setupIntent.payment_method) {
+			return {
+				paymentType: 'Stripe',
+				stripePublicKey,
+				stripePaymentType: 'StripeCheckout' as StripePaymentMethod,
+				paymentMethod: stripeIntentResult.setupIntent.payment_method as string,
+			};
+		}
+	}
+	return;
+};
+const getStripeExpressCheckoutPaymentFields = async (
+	stripeExpressCheckoutPaymentType: ExpressPaymentType | undefined,
+	stripe: Stripe | null,
+	stripeElements: StripeElements | null,
+	stripePublicKey: string,
+): Promise<RegularPaymentFields | undefined> => {
+	if (stripe && stripeElements) {
+		/** 1. Get a clientSecret from our server from the stripePublicKey */
+		const stripeClientSecret = await stripeCreateSetupIntentPrb(
+			stripePublicKey,
+		);
+
+		/** 2. Get the Stripe paymentMethod from the Stripe elements */
+		const { paymentMethod: stripePaymentMethod, error: paymentMethodError } =
+			await stripe.createPaymentMethod({
+				elements: stripeElements,
+			});
+
+		if (paymentMethodError) {
+			throw new FormSubmissionError(
+				'There was an issue with wallet.',
+				appropriateErrorMessage(paymentMethodError.decline_code ?? ''),
+			);
+			return;
+		}
+
+		/** 3. Get the setupIntent from the paymentMethod */
+		const { setupIntent, error: cardSetupError } =
+			await stripe.confirmCardSetup(stripeClientSecret, {
+				payment_method: stripePaymentMethod.id,
+			});
+
+		if (cardSetupError) {
+			throw new FormSubmissionError(
+				'There was an issue with wallet.',
+				appropriateErrorMessage(cardSetupError.decline_code ?? ''),
+			);
+			return;
+		}
+
+		const stripePaymentType: StripePaymentMethod =
+			stripeExpressCheckoutPaymentType === 'apple_pay'
+				? 'StripeApplePay'
+				: 'StripePaymentRequestButton';
+
+		/** 4. Pass the setupIntent through to the paymentFields sent to our /create endpoint */
+		return {
+			paymentType: 'Stripe',
+			paymentMethod: setupIntent.payment_method as string,
+			stripePaymentType,
+			stripePublicKey,
+		};
+	}
+	return;
+};
+
+export const getPaymentFieldsForPaymentMethod = async (
```

Comment:
very nicely pulled out of the `checkoutComponent`, am wondering a similar approach for `one-time-checkout` in future